lib/connection/result/row_stream.js

```diff
@@ -382,7 +382,9 @@ function externalizeRow(row, columns, mapColumnIdToExtractFnName, rowMode) {
   for (let index = 0, length = columns.length; index < length; index++) {
     const column = columns[index];
     const extractFnName = mapColumnIdToExtractFnName[column.getId()];
-    externalizedRow[isArrayRowMode ? index : column.getName()] = row[extractFnName](column.getId());
+    // wrap only numeric column names in quotes to prevent JS property autosorting and retain proper columm order
+    const columnName = isNaN(column.getName()) ? column.getName() : `'${column.getName()}'`; 
```

Comment:
does this mean that sometimes `column.getName()` is a number? It seems like the only way for this to be valid is if `column.getName()` actually returns a NaN, rather than being actually numeric?

Reply:
I can imagine 2 explanations here depending upon what `getName()` returns.
If it returns a string always, and the question is, "is this a numeric string?", then I think you want `isNaN(parseInt(column.getName()))` - then you only quote numeric strings.
If it actually sometimes returns a number, then I think you want `typeof column.getName() === "number"`

Reply:
wow, I stand corrected, `isNaN("test")` really does return true. what a crazy language